lints/lint_dnsname_bad_character_in_label.go

```diff
@@ -0,0 +1,67 @@
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"regexp"
+	"strings"
+)
+
+type DNSNameProperCharacters struct {
+	// Internal data here
+}
+
+func (l *DNSNameProperCharacters) Initialize() error {
+	return nil
+}
+
+func (l *DNSNameProperCharacters) CheckApplies(c *x509.Certificate) bool {
+	return true
+}
+
+func labelContainsBadCharacters(domain string) (bool, error) {
+	dnsLabelRegex := "^[A-Za-z0-9*_-]+$"
+	re, err := regexp.Compile(dnsLabelRegex)
```

Comment:
This should be in init. There's no reason you need to compile this for every check.